### ISSUE TITLE
js-controller 3.2.2  uncaught exception...

### ISSUE THREAD

**Issue by Ilovegym66** [iobroker-community-adapters/ioBroker.mihome-vacuum#210]
with js-controller 3.2.2, Node 14.15.1:

```
2021-01-09 10:34:13.113  - info: host.VMC123-iobroker Restart adapter system.adapter.mihome-vacuum.0 because enabled
2021-01-09 10:34:35.463  - info: linux-control.0 (444) getting data from octopi (192.168.4.13:22)
2021-01-09 10:34:42.133  - info: nuki.0 (1874) Bridge Info requested: http://192.168.0.30:8080/info?token=sl0ufi
2021-01-09 10:34:43.128  - info: host.VMC123-iobroker instance system.adapter.mihome-vacuum.0 started with pid 26792
2021-01-09 10:34:44.031  - info: mihome-vacuum.0 (26792) starting. Version 2.2.4 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v14.15.4, js-controller: 3.2.2
2021-01-09 10:34:44.110  - info: mihome-vacuum.0 (26792) Expert mode enabled, states created
2021-01-09 10:34:44.111  - info: mihome-vacuum.0 (26792) Cloud control disabled
2021-01-09 10:34:44.166  - info: mihome-vacuum.0 (26792) change states from State control.fan_power
2021-01-09 10:34:44.166  - info: mihome-vacuum.0 (26792) extend state mop for State control.fan_power
2021-01-09 10:34:44.171  - info: mihome-vacuum.0 (26792) connecting, this can take up to 10 minutes ...
2021-01-09 10:34:44.178  - error: mihome-vacuum.0 (26792) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 10:34:44.179  - error: mihome-vacuum.0 (26792) unhandled promise rejection: Not exists
2021-01-09 10:34:44.181  - error: mihome-vacuum.0 (26792) Error: Not exists
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:3010:38)
    at processImmediate (internal/timers.js:463:21)
2021-01-09 10:34:44.181  - error: mihome-vacuum.0 (26792) Not exists
2021-01-09 10:34:44.194  - info: mihome-vacuum.0 (26792) terminating
2021-01-09 10:34:44.194  - warn: mihome-vacuum.0 (26792) Terminated (UNCAUGHT_EXCEPTION): Without reason
2021-01-09 10:34:44.735  - warn: mihome-vacuum.0 (26792) Cannot get view: Error: Connection is closed.
2021-01-09 10:34:44.740  - info: mihome-vacuum.0 (26792) set nächster Timer: Nicht verfügbar
2021-01-09 10:34:44.741  - error: mihome-vacuum.0 (26792) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 10:34:44.741  - error: mihome-vacuum.0 (26792) unhandled promise rejection: States database not connected
2021-01-09 10:34:44.742  - error: mihome-vacuum.0 (26792) Error: States database not connected
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at Adapter.setState (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:5783:30)
    at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.mihome-vacuum/lib/timerManager.js:141:29)
    at processImmediate (internal/timers.js:463:21)
2021-01-09 10:34:44.743  - error: mihome-vacuum.0 (26792) States database not connected
2021-01-09 10:34:44.745  - error: mihome-vacuum.0 (26792) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 10:34:44.746  - error: mihome-vacuum.0 (26792) unhandled promise rejection: DB closed
2021-01-09 10:34:44.746  - error: mihome-vacuum.0 (26792) Error: DB closed
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at setObjectWithDefaultValue (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:1736:30)
2021-01-09 10:34:44.747  - error: mihome-vacuum.0 (26792) DB closed
2021-01-09 10:34:45.134  - info: nuki.0 (1874) Lock list requested: http://192.168.0.30:8080/list?token=sl0ufi
2021-01-09 10:34:45.476  - error: host.VMC123-iobroker instance system.adapter.mihome-vacuum.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
2021-01-09 10:34:45.477  - info: host.VMC123-iobroker Restart adapter system.adapter.mihome-vacuum.0 because enabled
```

**Comment by Apollon77** [comment]
Can it be that this error only happens when you restart the adapter? Or is the adapter not running at all?

**Comment by Ilovegym66** [comment]
is not running at all, no start possible.. :-(

**Comment by Grothesk242** [comment]
I second that issue: 

```
pi@raspberrypi:/opt/iobroker $ node -v
v14.15.4
pi@raspberrypi:/opt/iobroker $ npm -v
6.14.10
pi@raspberrypi:/opt/iobroker $ iobroker version
3.2.3
```


`2021-01-09 12:11:05.615  - info: host.raspberrypi "system.adapter.mihome-vacuum.0" enabled
2021-01-09 12:11:05.651  - info: host.raspberrypi instance system.adapter.mihome-vacuum.0 started with pid 14370
2021-01-09 12:11:07.652  - info: mihome-vacuum.0 (14370) starting. Version 2.1.1 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v14.15.4, js-controller: 3.2.3
2021-01-09 12:11:07.695  - info: mihome-vacuum.0 (14370) Expert mode disabled, states deleted
2021-01-09 12:11:07.697  - info: mihome-vacuum.0 (14370) Create state clean_home for controlling by cloud adapter
2021-01-09 12:11:07.737  - info: mihome-vacuum.0 (14370) New generation or new fw(3.5.8,004018) detected, create new states goto and zoneclean
2021-01-09 12:11:07.746  - info: mihome-vacuum.0 (14370) connecting, this can take up to 10 minutes ...
2021-01-09 12:11:07.773  - error: mihome-vacuum.0 (14370) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 12:11:07.773  - error: mihome-vacuum.0 (14370) unhandled promise rejection: Not exists
2021-01-09 12:11:07.777  - error: mihome-vacuum.0 (14370) Error: Not exists
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:3010:38)
    at processImmediate (internal/timers.js:463:21)
2021-01-09 12:11:07.777  - error: mihome-vacuum.0 (14370) Not exists
2021-01-09 12:11:07.780  - error: mihome-vacuum.0 (14370) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 12:11:07.781  - error: mihome-vacuum.0 (14370) unhandled promise rejection: Not exists
2021-01-09 12:11:07.782  - error: mihome-vacuum.0 (14370) Error: Not exists
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:3010:38)
    at processImmediate (internal/timers.js:463:21)
2021-01-09 12:11:07.783  - error: mihome-vacuum.0 (14370) Not exists
2021-01-09 12:11:07.803  - info: mihome-vacuum.0 (14370) terminating
2021-01-09 12:11:07.804  - warn: mihome-vacuum.0 (14370) Terminated (UNCAUGHT_EXCEPTION): Without reason
2021-01-09 12:11:08.222  - info: mihome-vacuum.0 (14370) set nächster Timer: Nicht verfügbar
2021-01-09 12:11:08.406  - error: host.raspberrypi instance system.adapter.mihome-vacuum.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
2021-01-09 12:11:08.407  - info: host.raspberrypi Restart adapter system.adapter.mihome-vacuum.0 because enabled
2021-01-09 12:11:31.547  - info: vodafone-speedtest.0 (14276) Vodafone-Speedtest finished with 73.715mbit download speed and 5.192mbit upload speed.
2021-01-09 12:11:31.556  - info: vodafone-speedtest.0 (14276) cleaned everything up...
2021-01-09 12:11:31.568  - info: vodafone-speedtest.0 (14276) Terminated (NO_ERROR): Without reason
2021-01-09 12:11:32.108  - info: host.raspberrypi instance system.adapter.vodafone-speedtest.0 terminated with code 0 (NO_ERROR)
2021-01-09 12:11:38.452  - info: host.raspberrypi instance system.adapter.mihome-vacuum.0 started with pid 14674
2021-01-09 12:11:41.127  - info: mihome-vacuum.0 (14674) starting. Version 2.1.1 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v14.15.4, js-controller: 3.2.3
2021-01-09 12:11:41.170  - info: mihome-vacuum.0 (14674) Expert mode disabled, states deleted
2021-01-09 12:11:41.172  - info: mihome-vacuum.0 (14674) Create state clean_home for controlling by cloud adapter
2021-01-09 12:11:41.207  - info: mihome-vacuum.0 (14674) New generation or new fw(3.5.8,004018) detected, create new states goto and zoneclean
2021-01-09 12:11:41.214  - info: mihome-vacuum.0 (14674) connecting, this can take up to 10 minutes ...
2021-01-09 12:11:41.229  - error: mihome-vacuum.0 (14674) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 12:11:41.230  - error: mihome-vacuum.0 (14674) unhandled promise rejection: Not exists
2021-01-09 12:11:41.235  - error: mihome-vacuum.0 (14674) Error: Not exists
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:3010:38)
    at processImmediate (internal/timers.js:463:21)
2021-01-09 12:11:41.236  - error: mihome-vacuum.0 (14674) Not exists
2021-01-09 12:11:41.238  - error: mihome-vacuum.0 (14674) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-01-09 12:11:41.238  - error: mihome-vacuum.0 (14674) unhandled promise rejection: Not exists
2021-01-09 12:11:41.239  - error: mihome-vacuum.0 (14674) Error: Not exists
    at Object.maybeCallbackWithError (/opt/iobroker/node_modules/iobroker.js-controller/lib/tools.js:2270:17)
    at Immediate.<anonymous> (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:3010:38)
    at processImmediate (internal/timers.js:463:21)
2021-01-09 12:11:41.240  - error: mihome-vacuum.0 (14674) Not exists
2021-01-09 12:11:41.260  - info: mihome-vacuum.0 (14674) terminating
2021-01-09 12:11:41.261  - warn: mihome-vacuum.0 (14674) Terminated (UNCAUGHT_EXCEPTION): Without reason
2021-01-09 12:11:41.692  - info: mihome-vacuum.0 (14674) set nächster Timer: Nicht verfügbar
2021-01-09 12:11:41.837  - error: host.raspberrypi instance system.adapter.mihome-vacuum.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
2021-01-09 12:11:41.838  - info: host.raspberrypi Restart adapter system.adapter.mihome-vacuum.0 because enabled`

**Comment by Apollon77** [comment]
it tries to delete an object which not exist

**Comment by Apollon77** [comment]
We will fix this in js-controller later ... delObject if not exists is kind of successfull :-)

**Comment by Apollon77** [comment]
fixed with js-controller 3.2.4
